### ISSUE TITLE
Login - Design UI for desktop

### ISSUE THREAD

**Issue by thisandagain** [mozilla/webmaker.org#1366]
Design UI / spec for mobile (webmaker-app). Depends on #1364

**Comment by cassiemc** [comment]
Initial signup screens for mobile: http://invis.io/N52EKLSB2
@vazquez could I get a prelim design review? Don't want to get too far along the aesthetic path if certain elements or ideas aren't working. Thanks!

**Comment by vazquez** [comment]
@cassiemc No problem, send me an invite to the InVision project so I can comment

**Comment by vazquez** [comment]
@cassiemc Left feebdack!

**Comment by thisandagain** [comment]
@cassiemc Looks like this is showing the desktop designs in a mobile frame. Was that intentional?

**Comment by cassiemc** [comment]
Nice catch @thisandagain, I'm on it. Fleshing out hundreds of screens today :)

**Comment by cassiemc** [comment]
Two choices here for desktop signup. 

A:

![signup-desktop-a](https://cloud.githubusercontent.com/assets/1186326/6623165/a0b3c19c-c8b7-11e4-93f3-5d4e82e7c160.jpg)

or B:

![signup-desktop-b](https://cloud.githubusercontent.com/assets/1186326/6623164/a0b2e5d8-c8b7-11e4-8534-ac3235adeb69.jpg)

I like B, but am torn on the logo.

**Comment by thisandagain** [comment]
:smile: :+1: :cake: Really digging the visual direction of B! Totally agree on the logo though ... wonder if we should just stick with our normal logo for now and revisit down the road? Happy to go with whatever you think is best given branding & styleguide process.

**Comment by cassiemc** [comment]
All the desktop screens, including some options toward the end for different suggested name treatments. http://invis.io/N52EKLSB2 

I'll post the suggested name treatments here too so we can choose to go forward with the feature or not, and any specific treatments. My instinct is to hold on this feature, come back to it when we can focus on the UX and UI of that feature alone. Thoughts @thisandagain – how important is it to the current flow?

A:
![signup-desktop-v3-suggested-names-a](https://cloud.githubusercontent.com/assets/1186326/6639911/12e61668-c96a-11e4-85e3-9c1ef3aa02c9.jpg)

B:
![signup-desktop-v3-suggested-names-b](https://cloud.githubusercontent.com/assets/1186326/6639912/12f0a9b6-c96a-11e4-80fa-7d10952f3216.jpg)

C:
![signup-desktop-v3-suggested-names-c](https://cloud.githubusercontent.com/assets/1186326/6639914/12fd860e-c96a-11e4-95f2-bdbcd6cb8845.jpg)

D:
![signup-desktop-v3-suggested-names-d](https://cloud.githubusercontent.com/assets/1186326/6639913/12f84658-c96a-11e4-93ee-884ddbc62e35.jpg)

**Comment by cassiemc** [comment]
Ugh just realized I posted this in the wrong bug (these are for desktop, not mobile). I am going to switch the titles around with bug https://github.com/mozilla/webmaker.org/issues/1365 to reflect what is actually being designed here. :)

**Comment by adamlofting** [comment]
It just crossed my mind that if this login is used on teach.*mozilla*.org, those user's may not actually have a concept of "Webmaker", depending on how the teach site get's branded in the long run... 

cc @hannahkane

**Comment by cassiemc** [comment]
Good point Adam. I think we'll have to look at it with that lens and decide if we want another more white label UI solution. I'm continuing to work on the UI for signin for desktop and mobile today.

**Comment by hannahkane** [comment]
Agree that this is the wrong copy and logo for the Teach site, though the UX is great.

**Comment by thisandagain** [comment]
Great. This is super helpful ... we can certainly find a way to vary branding & messaging by referring URL. @hannahkane can you spec some copy and we can get this requirement built into the spec? 

/cc @secretrobotron

**Comment by hannahkane** [comment]
Working on it as we speak.

**Comment by thisandagain** [comment]
@secretrobotron @cassiemc This requirement also impacts the design of the token and password reset transactional emails. We might want to make both more generic "Mozilla".

**Comment by hannahkane** [comment]
Copy deltas for Teach site version of login:

Replace "Build the web. Learn new skills." with "Teach the web. Learn new skills."
Replace "Free and open source - forever." with "Join our community of educators and activists."
"By signing up, I agree to Mozilla's Terms of Service and Privacy Policy" [replace "Webmaker's" with "Mozilla's"]
Also change the logo from Webmaker to Mozilla.

**Comment by cassiemc** [comment]
@hannahkane does this work? @vazquez for Design R? http://invis.io/W32GNX3V7

![signup-desktop-teach](https://cloud.githubusercontent.com/assets/1186326/6690846/4f6e9d1c-cc99-11e4-9e79-2fd7bb8604e6.jpg)

**Comment by vazquez** [comment]
Review done

**Comment by hannahkane** [comment]
+1

**Comment by thisandagain** [comment]
@cassiemc status update?

**Comment by cassiemc** [comment]
FYI we are not going to use the pink version since we don't need a separate design for Teach.

Here's the login screen for desktop but it is so f'in boring I am punching myself in the head. At least it is something to go by if devs need something to see. 

![login-desktop](https://cloud.githubusercontent.com/assets/1186326/6757237/fb7434f0-cf05-11e4-8aed-69c4fc5b7354.jpg)

**Comment by cassiemc** [comment]
I added a bunch more benchmarks to the login folder. One insight – there is usually a big difference in the way login and signup are treated visually; login is usually plainer, signup has a bit more fluff. https://drive.google.com/a/mozilla.com/folderview?id=0BzfN9sXlOXt1b2k0SGEteEVBSE0&usp=sharing

**Comment by cassiemc** [comment]
Ok, here's the latest after a few iterations with @k88hudson, thanks for that Kate. 
![login-desktop-v6](https://cloud.githubusercontent.com/assets/1186326/6759761/fdf146f6-cf18-11e4-8a89-134a3c592664.jpg)

**Comment by Pomax** [comment]
hm, that looks oddly similar to the original windows 98 default theme

**Comment by cassiemc** [comment]
Well, I think it's a little different from that... 

![windows-98](https://cloud.githubusercontent.com/assets/1186326/6760508/a31f5d9c-cf1f-11e4-81ec-8ca34dc38414.jpg)

Here is the latest (spacing adjusted somewhat):

![login-desktop-v7](https://cloud.githubusercontent.com/assets/1186326/6760524/cbc5c510-cf1f-11e4-8c5e-298382a6664d.jpg)

Curious to hear what @vazquez @sabrinang @flukeout think of the new direction given some of the UI work on the tool?

**Comment by k88hudson** [comment]
hahahahhahahahahhahahhahahhahahahhahahahhahahahahha

**Comment by cassiemc** [comment]
A couple options for the retrieve a token screen. I think I prefer the light blue as it doesn't overpower the login button. Not sure that the copy is right for the field – ie. the message says "key" and the field says "token". Thoughts @edrushka? 

(Note the header bar in these versions is also blue instead of a gradient - we will need to make some decisions Monday):

![login-desktop-7-check-your-email](https://cloud.githubusercontent.com/assets/1186326/6761141/51beb298-cf26-11e4-88b2-45ac3dcaf54a.jpg)
![login-desktop-7-check-your-email-green](https://cloud.githubusercontent.com/assets/1186326/6761142/56127910-cf26-11e4-9dfb-6113bfddd2b5.jpg)

**Comment by edrushka** [comment]
Definitely agree there should be consistency in terms. I prefer 'key'. Suggest something along the lines of:
Title: Retrieve your key
Subtitle: We've emailed your key to the address you provided.
Hint text: Paste your key here
Button text: Submit Key to Log In

@LucyeoH you mentioned that Login questions are the most frequently at the help address. Are there more specifics you can provide about where people get lost?

**Comment by alicoding** [comment]
@cassiemc can I have the assets of any the page that you have completed? mainly the logo and that background image if we are still going to use the same sign up page :)

**Comment by cassiemc** [comment]
@alicoding Can you pull the background image from the homepage? Same for the logo? Logo is here if you can't find it elsewhere: https://www.dropbox.com/sh/7nwgazjrlbh1x1l/AAAK-yKt6-DgCHQ_Yy2BjiSQa?dl=0

**Comment by alicoding** [comment]
Hmm I thought you have B/W logo like this one @cassiemc that's why I asked for that otherwise I will use what I have or the one you provided https://dl.dropboxusercontent.com/spa/n5v3bx9nnjkdpzf/p7d2oaw_.png

**Comment by cassiemc** [comment]
Oh I thought you meant for the sign up page, thanks for clarifying @alicoding. 

This logo treatment might have broader implications, would like to see what @HPaulJohnson and @thisandagain think before we implement?

**Comment by cassiemc** [comment]
On second thought, I am going to file this as a followup ticket as soon as @secretrobotron points me to the right repo. So n/m @thisandagain @HPaulJohnson – I will flag you on a different ticket (not for mvp).

**Comment by LucyeoH** [comment]
@edrushka oh god! Please don't ask people to retrieve a key! 

So every time someone logs in to Webmaker they need to retrieve a key from their email. At this point for reasons unknown many people do not receive this key, even if they have received it in the past. 

The only way to resolve this is for them to clear their cookies at which point they can try again and this time it will send them a key and they can login. I filed it in January but we haven't been able to find the exact break. @ScottDowne and I were able to replicate for a time but after we cleared the cookies on my computer we couldn't replicate again. All I know is that for lots and lots of people this isn't working well. I have received over 22 help requests about this issue alone. https://github.com/mozilla/login.webmaker.org/issues/313#issuecomment-71516056

**Comment by edrushka** [comment]
@LucyeoH thanks very much for this. @cassiemc let me know how you'd like to proceed with the copy. Changing the login process seems out of scope for this ticket.

**Comment by LucyeoH** [comment]
@edrushka totally get it!

**Comment by cassiemc** [comment]
@LucyeoH @edrushka This screen is all about transitioning people who haven't set a password yet to choosing a password, but we will need them to retrieve a key one last time for this to work, as far as I know. We are moving away from 'keys' and that is the whole reason for us redesigning login :) I've broken this ticket down into several others - we can continue this particular conversation in https://github.com/mozilla/id.webmaker.org/issues/28

All other design issues can be found in this list: https://github.com/mozilla/id.webmaker.org/issues?q=is%3Aopen+is%3Aissue+label%3Adesign 

Am going to close this ticket now.

@thisandagain @secretrobotron @jbuck do we need to have another look at this or is what I said above correct?

**Comment by thisandagain** [comment]
Yup! :+1: That's right on @cassiemc. Thanks for breaking this up into smaller issues within the `id` repo.

**Comment by Pomax** [comment]
@cassiemc I was more reminded of the title bar in in standard installs like ![win98 deliciousness](http://3.bp.blogspot.com/_ieHehyhaqMo/TSaDjTbpTLI/AAAAAAAAACE/uBD_MSv20ro/s1600/windows-98.jpg) =)

**Comment by cassiemc** [comment]
@Pomax I see your point :)
